MonsterTrainModdingAPI/Managers/CustomTriggerManager.cs

```diff
@@ -0,0 +1,165 @@
+﻿using System;
+using System.Collections;
+using System.Collections.Generic;
+using System.Text;
+using MonsterTrainModdingAPI.Enums.MTTriggers;
+using UnityEngine;
+
+namespace MonsterTrainModdingAPI.Managers
+{
+    public class CustomTriggerManager
+    {
+        private const int TriggerGUIDOffset = 100;
+        private static int NumTriggers = Enum.GetNames(typeof(CharacterTriggerData.Trigger)).Length + TriggerGUIDOffset;
+
+        /// <summary>
+        /// Gets a New GUID and sets up a Localization Key of [Trigger_TypeName]
+        /// </summary>
+        /// <typeparam name="T">The type of the MTTrigger attempting</typeparam>
+        /// <returns></returns>
+        public static int GetNewGUID<T>() where T : IMTCharacterTrigger
+        {
+            return GetNewGUID<T>("Trigger_" + typeof(T).Name);
+        }
+        /// <summary>
+        /// Gets a New GUID and sets up a Localization key
+        /// </summary>
+        /// <typeparam name="T"></typeparam>
+        /// <param name="LocalizationKey"></param>
+        /// <returns></returns>
+        public static int GetNewGUID<T>(string LocalizationKey) where T : IMTCharacterTrigger
```

Comment:
This function does more than one thing. I'd generally advise against that, but if this is a particularly common use case (and it sounds like it is?) then it's okay. The name should reflect that, though. This is clearly doing something more than just getting a new GUID.